acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -504,14 +504,14 @@ def start_puppet_non_blocking_request(broker, targets, environment = 'production
 end
 
 def check_non_blocking_response(broker, identity, transaction_id, max_retries, query_interval, &block)
-  run_result = nil
+  rpc_action_status = nil
   query_attempts = 0
-  until (query_attempts == max_retries || run_result) do
+  until (query_attempts == max_retries || rpc_action_status) do
     query_responses = rpc_blocking_request(broker, [identity],
                                            'status', 'query', {:transaction_id => transaction_id})
     action_result = query_responses[identity][:data]['results']
     assert(action_result, "Response to status query was an error: #{query_responses[identity][:data]}")
-    if (action_result.has_key?('stdout') && (action_result['stdout'] != ""))
+    if action_result.has_key?('exitcode')
       rpc_action_status = action_result['status']
       run_result = action_result['stdout']
```

Comment:
It seems we could just `break` out of the loop here, which would allow us to drop the testing of the `rpc_action_status` in the condition of the loop and the `if` condition (which still tests the possibly empty stdout) around the sleep below.

Reply:
Actually I'd discard the loop condition altogether in favor of:
```ruby
loop do
    ...
    if action_result.has_key?('exitcode')
        rpc_action_status = action_result['status']
        run_result = action_result['stdout']
        break
    end
    query_attempts += 1
    break if query_attempts == max_retries
    sleep query_interval
end
```

Reply:
Looks better, updated.